### PR TITLE
Added undici to lookup.json

### PR DESCRIPTION
**Pull request by mcollina**
Adding [undici](https://github.com/nodejs/undici).

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [x] contribution guidelines followed
      [here](https://github.com/nodejs/citgm/blob/HEAD/CONTRIBUTING.md)

### CODE REVIEW

File: lib/lookup.json [hunk 1]
```diff
@@ -507,6 +507,9 @@
     "skip": "win32",
     "maintainers": "jashkenas"
   },
+  "undici": {
+    "maintainers": ["mcollina", "ronag"]
```
Comment (BridgeAR) [inline lib/lookup.json:511]: Seems like it requires the prefix `v`, or am I mistaken?

**Comment by targos** [comment]
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/134/

File: lib/lookup.json (hunk 1 above)
Comment (mcollina) [reply lib/lookup.json:511]: how could I now? how is the prefix picked up? Is it in the github tag?

**Comment by targos** [comment]
I assume it should be skipped on v10 ? Fails because of the use of ESM in tests.

**Comment by mcollina** [comment]
v10 is supported https://github.com/nodejs/undici/blob/master/.github/workflows/nodejs.yml

**Comment by targos** [comment]
Citgm executes `npm run test` by default. Should we setup another script?

**Comment by mcollina** [comment]
I don't understand what is the failure, tests are passing with Node v10.

**Comment by targos** [comment]
https://ci.nodejs.org/job/citgm-smoker-nobuild/1004/nodes=fedora-latest-x64/testReport/junit/(root)/citgm/undici_v3_0_0/

The failure is in test/esm-wrapper

**Comment by mcollina** [comment]
Dang, thanks for spotting this. We run a different task on and it got spuriously broken. I've fixed it in https://github.com/nodejs/undici/pull/511

**Comment by mcollina** [comment]
v3.1.0 is out and it should work now. Can you check @targos?

**Comment by targos** [comment]
Sure, CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/135/console

**Review by targos — APPROVED** [review]

File: lib/lookup.json (hunk 1 above)
Comment (targos) [reply lib/lookup.json:511]: It is usually not necessary to specify the tag prefix. If the npm package is published from a git repo, npm sets the git SHA in the package metadata and we use that to download the tarball.

**Comment by targos** [comment]
The only failure is https://github.com/nodejs/build/issues/2515, LGTM.

**Review by richardlau — APPROVED** [review]

**Pull request merged**